qiling/os/posix/syscall/wait.py

```diff
@@ -10,12 +10,19 @@
 from qiling.os.filestruct import *
 from qiling.os.posix.const_mapping import *
 from qiling.exception import *
+from qiling.os.macos.const import ECHILD
 from qiling.utils import *
 
+
 def ql_syscall_wait4(ql, wait4_pid, wait4_wstatus, wait4_options, wait4_rusage, *args, **kw):
-    spid, status, rusage = os.wait4(wait4_pid, wait4_options)
-    if wait4_wstatus != 0:
-        ql.mem.write(wait4_wstatus, ql.pack32(status))
-    regreturn = spid
-    ql.nprint("wait4(%d, %d) = %d"% (wait4_pid, wait4_options, regreturn))
+    wait4_pid = ql.unpacks(ql.pack(wait4_pid))  # convert to signed
+    try:
+        spid, status, rusage = os.wait4(wait4_pid, wait4_options)
+        if wait4_wstatus != 0:
+            ql.mem.write(wait4_wstatus, ql.pack32(status))
+        regreturn = spid
+    except ChildProcessError:
```

Comment:
exception ChildProcessError is a subclass of OSError which can be thrown by os.wait4.
It is raised when an operation on a child process failed. Corresponds to errno ECHILD.
It is documented [here](https://docs.python.org/3/library/exceptions.html).

Reply:
I don't think this exception will work here. Infract, we dont even need it ?

Reply:
We need to catch the Exception because it was thrown by python when executing a binary. The binary called wait4 with a non existing PID. We ran qiling on Ubuntu 20.4 and Kali and on both python stopped with the ChildProcessError Exception instead of passing down the error to the simulated binary.

Reply:
Could it be just pack or pack32 ?